app/javascript/listings/listingDashboard.jsx

```diff
@@ -1,232 +1,211 @@
-  }
-
-  render() {
-    const {
-      listings,
-      orgListings,
-      userCredits,
-      orgs,
-      selectedListings,
-      filter,
-      sort,
-    } = this.state;
-
-    const isExpired = (listing) =>
-      listing.bumped_at && !listing.published
-        ? (Date.now() - new Date(listing.bumped_at.toString()).getTime()) /
-            (1000 * 60 * 60 * 24) >
-          30
-        : false;
-    const isDraft = (listing) =>
-      listing.bumped_at ? !isExpired(listing) && !listing.published : true;
-
-    const filterListings = (listingsToFilter, selectedFilter) => {
-      if (selectedFilter === 'Draft') {
-        return listingsToFilter.filter((listing) => isDraft(listing));
-      }
-      if (selectedFilter === 'Expired') {
-        return listingsToFilter.filter((listing) => isExpired(listing));
-      }
-      if (selectedFilter === 'Active') {
-        return listingsToFilter.filter((listing) => listing.published === true);
-      }
-      return listingsToFilter;
-    };
 
-    const customSort = (a, b) => {
-      if (a[sort] === null) {
-        return 1;
-      }
-      if (b[sort] === null) {
-        return -1;
-      }
-      if (a[sort] > b[sort]) {
-        return -1;
-      }
+    setListings(listings);
+    setOrgs(JSON.parse(container.dataset.orgs));
+    setOrgListings(JSON.parse(container.dataset.orglistings));
+    setUserCredits(container.dataset.usercredits);
+  }, []);
+
+  const isExpired = (listing) =>
+    listing.bumped_at && !listing.published
+      ? (Date.now() - new Date(listing.bumped_at.toString()).getTime()) /
+          (1000 * 60 * 60 * 24) >
+        30
+      : false;
+
+  const isDraft = (listing) =>
+    listing.bumped_at ? !isExpired(listing) && !listing.published : true;
+
+  const filterListings = (listingsToFilter, selectedFilter) => {
+    if (selectedFilter === 'Draft') {
+      return listingsToFilter.filter((listing) => isDraft(listing));
+    }
+    if (selectedFilter === 'Expired') {
+      return listingsToFilter.filter((listing) => isExpired(listing));
+    }
+    if (selectedFilter === 'Active') {
+      return listingsToFilter.filter((listing) => listing.published === true);
+    }
+    return listingsToFilter;
+  };
+
+  const customSort = (a, b) => {
+    if (a[sort] === null) {
       return 1;
-    };
+    }
+    if (b[sort] === null) {
+      return -1;
+    }
+    if (a[sort] > b[sort]) {
+      return -1;
+    }
+    return 1;
+  };
 
-    const showListings = (
-      selected,
-      userListings,
+  const showListings = (
+    selected,
+    userListings,
+    organizationListings,
+    selectedFilter,
+  ) => {
+    let displayedListings;
+    if (selected === 'user') {
+      displayedListings = filterListings(userListings, selectedFilter).sort(
+        customSort,
+      );
+      return displayedListings.map((listing) => (
+        <ListingRow listing={listing} key={listing.id} />
+      ));
+    }
+    displayedListings = filterListings(
```

Comment:
(not blocking) - if we wanted to, we could refactor this further - I don't think we need the `let displayedListings`, since we always return in the `if (selected === 'user')` block and so don't need to capture its listings in a variable 🤷‍♀️ 

It's not new to this PR though, just if you fancied it 😄